ckanext/switzerland/helpers/format_utils.py

```diff
@@ -94,11 +100,44 @@ def prepare_formats_for_index(resources, format_mapping):
     return list(formats)
 
 
-# all formats that need to be mapped have to be entered in the mapping.yaml
-def _map_to_valid_format(resource_format, format_mapping):
-    resource_format_lower = resource_format.lower()
-    for key, values in format_mapping.iteritems():
-        if resource_format_lower in values:
-            return key
-    else:
-        return None
+def _map_to_valid_format(format, try_to_clean=False):
+    """check whether the format if in the mapping:
+    either as a key or as a value or if it can be derived
+    after cleaning the input format string"""
+    if format in format_mapping.keys():
+        return format
+
+    if format in reverse_format_mapping.keys():
+        return reverse_format_mapping[format]
+    if try_to_clean:
+        cleaned_format = _get_cleaned_format(format)
+        if cleaned_format in reverse_format_mapping.keys():
+            return reverse_format_mapping[cleaned_format]
+    return ""
+
+
+def _get_cleaned_media_type(media_type):
+    """clean the media type"""
+    cleaned_media_type = media_type.split('/')[-1].lower()
+    if cleaned_media_type:
+        return cleaned_media_type
+    return ""
+
+
+def _get_cleaned_format(format):
+    """clean the format"""
+    if format:
+        cleaned_format = format.split('/')[-1].lower()
+        return cleaned_format
+    return ""
```

Comment:
Functionally, the code looks correct to me now. :) However, these two functions do essentially the same thing, and I feel it's quite confusing to have them both here. Could you use just `_get_cleaned_format` and get rid of `_get_cleaned_media_type`?

Reply:
well spotted. I will fix this.